pkg/values/big_int.go

```diff
@@ -23,10 +24,15 @@ func (b *BigInt) proto() *pb.Value {
 }
 
 func (b *BigInt) Unwrap() (any, error) {
-	return b.Underlying, nil
+	bi := &big.Int{}
+	return bi, b.UnwrapTo(bi)
 }
 
 func (b *BigInt) UnwrapTo(to any) error {
+	if b == nil {
```

Comment:
Should you also check if b.Underlying is nil? (same for other types below)

Reply:
For bigInt yes, good catch :) For the others, I've added tests to show that it isn't necessary